pontoon/base/models.py

```diff
@@ -2585,6 +2585,29 @@ def reset_active_translation(self, locale, plural_form=None):
         else:
             return Translation()
 
+    def reset_term_translation(self, locale):
+        """
+        When translation in the "Terminology" project changes, update the corresponding
+        TermTranslation:
+        - If approved translation exists, use it as TermTranslation
+        - If approved translation doesn't exist, remove any TermTranslation instance
+
+        This method is also executed in the process of deleting a term translation,
+        because it needs to be rejected first, which triggers the call to this
+        function.
+        """
+        approved_translations = self.translation_set.filter(
+            locale=locale, approved=True
+        )
+        term = self.term
+
+        if approved_translations:
+            term_translation, _ = term.translations.get_or_create(locale=locale)
+            term_translation.text = approved_translations[0].string
```

Comment:
[question] Is there a moment when there's more than 1 approved translation for a given locale? Maybe calling `.get()` could be more explicit here (?).